.github/workflows/ci-github-actions.yml

```diff
@@ -0,0 +1,24 @@
+# Continuous integration handling for GitHub Actions workflows
+name: ci-github-actions
+
+on:
+  pull_request:
+    paths:
+      - .github/workflows/*.yml
+
+permissions:
+  contents: read
+
+jobs:
+  actionlint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - id: go-version
+        # Reference: https://github.com/actions/setup-go/issues/23
+        run: echo "::set-output name=version::$(cat ./.go-version)"
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ steps.go-version.outputs.version }}
+      - run: go install github.com/rhysd/actionlint/cmd/actionlint@latest
+      - run: actionlint
```

Comment:
Nice! Maybe they'll bundle the various `actionlint` install/execution bits into its own GitHub Action at some point -- I'd be a little worried about manually maintaining a point-in-time copied JSON file with confusing regular expressions!

Reply:
I definitely agree! If you are intending to reuse the same workflows across multiple Terraform repos, it may be advantageous to have a shared public repo with [reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows) defined. That'd let you more cleanly adhere to the DRY principle, with each "consuming" repo having workflows that look like so:


```yml
# Continuous integration handling for GitHub Actions workflows
name: ci-github-actions

on:
  pull_request:
    paths:
      - .github/workflows/*.yml

permissions:
  contents: read

jobs:
  reusable_workflow_job:
    runs-on: ubuntu-latest
    steps:
      - uses: hashicorp/<<insert_repo_name_here>>/.github/workflows/actionlint.yml@main
```

Just tossing an idea out there 🙂